internal/gogrep/instructions.go

```diff
@@ -0,0 +1,107 @@
+package gogrep
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+)
+
+type bitmap64 uint64
+
+func (m bitmap64) IsSet(pos int) bool {
+	return m&(1<<pos) != 0
+}
+
+type valueKind uint8
+
+const (
+	emptyValue   valueKind = iota
+	stringValue            // Extra values only; value is stored in program.strings
+	ifaceValue             // Extra values only; value is stored in program.ifaces
+	tokenValue             // token.Token
+	chandirValue           // ast.CharDir
+)
+
+type program struct {
+	insts   []instruction
+	strings []string
```

Comment:
Probably `strings` is ok as field name, but not so as a variable. (comment just for attention)